File: Dagger2Ex/app/src/main/java/expencive/vk/com/dagger2ex/car/DieselEngine.java
package expencive.vk.com.dagger2ex.car;

import android.util.Log;

import javax.inject.Inject;

public class DieselEngine implements Engine {
    private static final String TAG = "DieselEngine";

    private int horsePower;


    public DieselEngine(int horsePower) {
        this.horsePower = horsePower;
    }

    @Override
    public void start() {

        Log.d(TAG, "start: Diesel Engine.........................................................." +
                "Horsepower" + horsePower);

    }

}

File: DaggerKotlin/app/src/main/java/vk/expencive/daggerkotlin/car/DieselEngine.kt
package vk.expencive.daggerkotlin.car

import android.util.Log
import javax.inject.Inject

class DieselEngine (private val horsePower: Int): Engine {
    private val TAG = "DieselEngine"



    override fun wrom() {
        Log.d(TAG, "drive: Diesel rrrrrrrrrrrrrrrr: horsePower ${horsePower} ")
    }
}